src/main/java/org/openlmis/integration/dhis2/service/DhisDataService.java

```diff
@@ -122,7 +126,46 @@ public ArrayList<DhisDataset> getDhisDatasets(String serverUrl, String username,
       throw new ResponseParsingException(
               MessageKeys.ERROR_EXTERNAL_API_RESPONSE_BODY_UNABLE_TO_PARSE, ex);
     }
+  }
 
+  /**
+   * Get all organisational units for a given server from DHIS2 API.
+   *
+   * @param serverUrl Url of the dhis2 server.
+   * @param username  Name of the specific user.
+   * @param password  User password.
+   * @return the {@link OrganisationUnit} list.
+   */
+  public ArrayList<OrganisationUnit> getDhisOrgUnits(String serverUrl, String username,
+                                                String password) {
+    String token = authService.obtainAccessToken(username, password, serverUrl);
+
+    RequestParameters params = RequestParameters
+            .init()
+            .set("paging", "false");
+
+    ResponseEntity<LinkedHashMap<String, ArrayList<Object>>> orgUnitResponse;
+    try {
+      orgUnitResponse = restTemplate.exchange(
+              createUri(serverUrl + API_ORG_UNITS_URL, params),
+              HttpMethod.GET,
+              createEntity(token, API_TOKEN),
+              new ParameterizedTypeReference
+                      <LinkedHashMap<String, ArrayList<Object>>>() {}
+      );
+
+      ArrayList<?> orgUnits
+              = orgUnitResponse.getBody().get("organisationUnits");
+      return (ArrayList<OrganisationUnit>) orgUnits;
+    } catch (HttpClientErrorException ex) {
+      throw new RestOperationException(
+              MessageKeys.ERROR_EXTERNAL_API_CLIENT_REQUEST_FAILED, ex);
+    } catch (RestClientException ex) {
+      throw new RestOperationException(MessageKeys.ERROR_EXTERNAL_API_CONNECTION_FAILED, ex);
+    } catch (NullPointerException ex) {
```

Comment:
As the frontend side is blocked because of this ticket, I will move resolving this issue to https://openlmis.atlassian.net/browse/SELV3-594 as there was already a request to generify data fetching in whole integration